android/src/main/java/com/theweflex/react/WeChatModule.java

```diff
@@ -253,84 +243,72 @@ private static Uri getResourceDrawableUri(Context context, String name) {
                 .build();
     }
 
-    private Drawable _createDrawable(CloseableReference<CloseableImage> image) {
-        Preconditions.checkState(CloseableReference.isValid(image));
-        CloseableImage closeableImage = image.get();
-        if (closeableImage instanceof CloseableStaticBitmap) {
-            CloseableStaticBitmap closeableStaticBitmap = (CloseableStaticBitmap) closeableImage;
-            BitmapDrawable bitmapDrawable = new BitmapDrawable(
-                    getReactApplicationContext().getResources(),
-                    closeableStaticBitmap.getUnderlyingBitmap());
-            if (closeableStaticBitmap.getRotationAngle() == 0 ||
-                    closeableStaticBitmap.getRotationAngle() == EncodedImage.UNKNOWN_ROTATION_ANGLE) {
-                return bitmapDrawable;
-            } else {
-                return new OrientedDrawable(bitmapDrawable, closeableStaticBitmap.getRotationAngle());
-            }
-        } else if (closeableImage instanceof CloseableAnimatedImage) {
-            return Fresco.getImagePipelineFactory().getAnimatedDrawableFactory().create(
-                    ((CloseableAnimatedImage) closeableImage).getImageResult());
-        } else {
-            throw new UnsupportedOperationException("Unrecognized image class: " + closeableImage);
+    private void _share(final int scene, final ReadableMap data, final Bitmap thumbImage, final Callback callback) {
+        if (!data.hasKey("type")) {
+            callback.invoke(INVALID_ARGUMENT);
+            return;
         }
-    }
+        String type = data.getString("type");
 
-    private Bitmap _drawable2Bitmap(Drawable drawable) {
-        if (drawable instanceof BitmapDrawable) {
-            return ((BitmapDrawable) drawable).getBitmap();
-        } else if (drawable instanceof NinePatchDrawable) {
-            Bitmap bitmap = Bitmap
-                    .createBitmap(
-                            drawable.getIntrinsicWidth(),
-                            drawable.getIntrinsicHeight(),
-                            drawable.getOpacity() != PixelFormat.OPAQUE ? Bitmap.Config.ARGB_8888
-                                    : Bitmap.Config.RGB_565);
-            Canvas canvas = new Canvas(bitmap);
-            drawable.setBounds(0, 0, drawable.getIntrinsicWidth(),
-                    drawable.getIntrinsicHeight());
-            drawable.draw(canvas);
-            return bitmap;
+        WXMediaMessage.IMediaObject mediaObject = null;
+        if (type.equals("news")) {
+            mediaObject = _jsonToWebpageMedia(data);
+        } else if (type.equals("text")) {
+            mediaObject = _jsonToTextMedia(data);
+        } else if (type.equals("imageUrl") || type.equals("imageResource")) {
```

Comment:
Hi @yorkie , In javascript, imageUrl will be images from web, It's url start with "http(s)", and the url of Image from static resource(assets) starts with "asset://", Their conceptual is not identical, even their implement code are same.

Please review my example code:https://github.com/weflex/react-native-wechat/blob/master/README.md#sharetotimelinedata.

This example code works fine on both iOS and Android now.

I will improve the official Example of our plugin.